app/javascript/packs/containers/ReviewModeContainer.jsx

```diff
@@ -7,10 +7,89 @@ import React from 'react';
 
 import QuestionContainer from './QuestionContainer';
 
+import { getAnswerForQuestionAndBuilding } from '../selectors/answersSelector';
+import { getDependentQuestionsForOption } from "../selectors/questionsSelector";
 import { connect } from 'react-redux';
 import { getQuestionsByBuilding } from '../selectors/questionsSelector';
 
+import { post, patch } from '../fetch/requester';
+
+async function postDelegations(delegations, callback) {
+  var body = {
+    "delegations": delegations,
+  };
+
+  try {
+    let resp = await post('/api/delegations', body);
+    callback(true);
+  } catch (error) {
+    callback(false);
+  }
+}
+
 class ReviewModeContainer extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      status_string: "Delegations have not been saved yet.",
+    }
+  }
+
+  // recursively find all dependent questions, return a list
+  getDependentQuestion(parentQuestion) {
+    var potentialChildren = [].concat(
+        ...Object.values(parentQuestion.options).map((option) => {
+          return this.props.getDependentQuestion(option.id, parentQuestion.question_type);
+        })
+    );
+
+    if (potentialChildren.length == 0) {
+      return [];
+    } else {
+      return potentialChildren.concat(potentialChildren.map(this.getDependentQuestion));
+    }
+  }
+
+  // called when delegation should be submitted
+  // should synchronously submit delegations since user expects success
```

Comment:
Hmm, not sure why this says synchronous since this calls `postDelegations`, which is async?

Reply:
My comment here means I think it should be done synchronously instead of async.

When the user click the submit button it really means he decides it's over, and I suppose a synchronous submission is more natural? That being said, it doesn't have to be sync in js, we just need to treat it differently than saving answer for certain question, during which the user has moved on.

Reply:
> I think it should be done synchronously instead of async.

Do you mean this from a UI perspective? - there isn't really a way to make this network request in a synchronous way. 

One last change I might consider is using the `async/await` pattern you've used previously instead of a callback (using both may be confusing and not consistent with how we normally handle async calls). You've already specified `postDelegations` as an async function, so why not try/catch directly here? Alternatively, you may return a value out of the async function and `await` it here.